src/main/java/uk/gov/ida/matchingserviceadapter/validators/FirstNameToComparator.java

```diff
@@ -0,0 +1,18 @@
+package uk.gov.ida.matchingserviceadapter.validators;
+
+import uk.gov.ida.saml.core.domain.TransliterableMdsValue;
+
+import java.util.Comparator;
+
+public class FirstNameToComparator implements Comparator<TransliterableMdsValue> {
```

Comment:
I think you can replace this with a one liner:
`Comparator.<TransliterableMdsValue>comparing(TransliterableMdsValue::getTo, Comparator.nullsFirst(Comparator.reverseOrder()))`